src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -875,6 +876,18 @@ void RegisterAppInterfaceRequest::SendRegisterAppInterfaceResponseToMobile(
   // Default HMI level should be set before any permissions validation, since it
   // relies on HMI level.
   application_manager_.OnApplicationRegistered(application);
+
+  if (application->is_remote_control_supported()) {
+    auto add_rc_extension = [application](plugin_manager::RPCPlugin& plugin) {
+      if (plugin.PluginName() ==
```

Comment:
@IGapchuk Why did you use it for rc_rpc_plugin ONLY?

Reply:
@AByzhynar On the current implementation, all extensions added to an application during running `ApplicationManagerImpl::RegisterApplication` method that called on the 377 line. But at that moment, the application is not marked as `remote_control_supported` and RCPlugin can't add RCExtension to the application. So we do that in this place when default HMI level will be set and the default window will be added.

Reply:
Sounds like a hack. Requires additional comment in the code.

Reply:
Do we have an option to mark application rc before all plugins extensions will be added?